### DISCUSSION TITLE
Best practices for reducing local/remote gradle cache misses during local development

### DISCUSSION THREAD

**Discussion by kyhule** [ReactiveCircus/app-versioning#16] [General]
By now, most are probably aware they should [use static build config values with your debug build](https://developer.android.com/studio/build/optimize-your-build#use_static_build_properties), but if your project takes advantage of remote gradle cache, you also do not want to make things so static (i.e. hardcode `versionCode` to 1) that you have unnecessary cache misses even when building an up to date checkout of the default branch with a clean working tree.

I'm curious to hear what best practices and patterns people use to optimize build cache when using this plugin.

**Comment by ychescale9** [comment]
Are you getting any unexpected cache misses with this plugin?

The versionCode and versionName generation are done in the `generatedAppVersionInfoFor<BuildVarient` task which is both incremental and cacheable as the task uses the project's `.git/refs` for tracking whether version info need to be recomputed due to changes to git tags.

Can you please explain what issues you're running into and what's your expectation?
